### PR TITLE
ENH move PSF center to object location

### PR DESCRIPTION
**Pull request by beckermr**
This should fix our size offset bugggggggz.

cc @esheldon @rmjarvis

### CODE REVIEW

File: desmeds/desdm_maker.py [hunk 1]
```diff
@@ -1088,7 +1090,26 @@ def get_center(self, row, col):
...
+        # it returns the central pixel of the image
+        # https://github.com/rmjarvis/Piff/blob/releases/1.2/piff/psf.py#L177
+        col_cen = int(np.ceil(col - (0.5 if sa[1] % 2 == 1 else 0)))
+        row_cen = int(np.ceil(row - (0.5 if sa[0] % 2 == 1 else 0)))
+
+        # these are the offset of the PSF position from the central pixel
+        dcol = col - col_cen
+        drow = row - row_cen
+
+        # now we add those offsets to the offset from the central pixel
```
Comment (beckermr) [inline desmeds/desdm_maker.py:1104]:
```suggestion
        # now we add those offsets to the central pixel
        # to get the cutout location of the PSF
```

File: desmeds/desdm_maker.py [hunk 2]
```diff
@@ -1088,7 +1090,27 @@ def get_center(self, row, col):
...
+        row_cen = int(np.ceil(row - (0.5 if sa[0] % 2 == 1 else 0)))
+
+        # these are the offset of the PSF position from the central pixel
+        dcol = col - col_cen
+        drow = row - row_cen
+
+        # now we add those offsets to the central pixel
+        # to get the cutout location of the PSF
+        # galsim rounds up for even images
+        row_cutout = ((sa[0] - 1)/2 if sa[0] % 2 == 1 else sa[0]/2) + drow
```
Comment (beckermr) [inline desmeds/desdm_maker.py:1107]:
```suggestion
        # these are zero-indexed
        row_cutout = ((sa[0] - 1)/2 if sa[0] % 2 == 1 else sa[0]/2) + drow
```

File: desmeds/desdm_maker.py [hunk 3]
```diff
@@ -1088,7 +1090,28 @@ def get_center(self, row, col):
...
+        # to get the cutout location of the PSF
+        # galsim rounds up for even images
+        # these are zero-indexed
+        row_cutout = ((sa[0] - 1)/2 if sa[0] % 2 == 1 else sa[0]/2) + drow
+        col_cutout = ((sa[1] - 1)/2 if sa[1] % 2 == 1 else sa[1]/2) + dcol
+
+        return np.array([
+            row_cutout,
+            col_cutout,
+        ])
```
Comment (rmjarvis) [inline desmeds/desdm_maker.py:1114]: I'm not sure if you have access to the image that Piff made at the point when you are calling this function.  But this whole result should just end up being (row - image.bounds.ymin, col - image.bounds.xmin).  Maybe that's fodder for a unit test at least...

Comment (beckermr) [reply desmeds/desdm_maker.py:1114]: Good point. The image isn't around which is why I am going to so much trouble here. I will check against the easy way.

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
+name: tests
+
+on:
+  push:
+    branches:
+      - main
```
Comment (beckermr) [inline .github/workflows/tests.yml:6]:
```suggestion
      - master
```

**Review by rmjarvis — APPROVED** [review]
LGTM

**Pull request merged**